tests/Unit/Chat/OpenAIChatTest.php

```diff
@@ -20,6 +20,29 @@
     expect(isset($chat))->toBeTrue();
 });
 
+it('can process system, user, assistant and functionResult messages', function () {
```

Comment:
While the integration test seems really useful, I don't see much value in this unit test. Moreover I think it does not represent an example of real usage of the function, so it could be misleading.